src/GitVersion.Core/VersionCalculation/VariableProvider.cs

```diff
@@ -2,64 +2,80 @@
+            && currentCommitTaggedVersion is null;
         if (isContinuousDeploymentMode)
         {
             semanticVersion = new SemanticVersion(semanticVersion);
             // Continuous Deployment always requires a pre-release tag unless the commit is tagged
             if (!semanticVersion.PreReleaseTag.HasTag())
             {
-                var label = configuration.GetBranchSpecificLabel(semanticVersion.BuildMetaData.Branch, null);
-                semanticVersion.PreReleaseTag.Name = label ?? string.Empty;
-                if (semanticVersion.PreReleaseTag.Name.IsNullOrEmpty())
+                // TODO: Why do we manipulating the semantic version here in the VariableProvider? The method name is GET not MANIPULATE.
+                // What is about the separation of concern and single-responsibility principle?
+                preReleaseTagName = configuration.GetBranchSpecificLabel(semanticVersion.BuildMetaData.Branch, null);
+                if (preReleaseTagName.IsNullOrEmpty())
                 {
-                    // TODO: Why do we manipulating the semantic version here in the VariableProvider? The method name is GET not MANIPULATE.
-                    // What is about the separation of concern and single-responsibility principle?
-                    semanticVersion.PreReleaseTag.Name = configuration.Label ?? string.Empty;
+                    preReleaseTagName = configuration.Label ?? string.Empty;
                 }
             }
         }
 
         // Evaluate tag number pattern and append to prerelease tag, preserving build metadata
         var appendTagNumberPattern = !configuration.LabelNumberPattern.IsNullOrEmpty() && semanticVersion.PreReleaseTag.HasTag();
-        if (appendTagNumberPattern)
+        if (appendTagNumberPattern && semanticVersion.BuildMetaData.Branch != null && configuration.LabelNumberPattern != null)
         {
-            if (semanticVersion.BuildMetaData.Branch != null && configuration.LabelNumberPattern != null)
+            var match = Regex.Match(semanticVersion.BuildMetaData.Branch, configuration.LabelNumberPattern);
+            var numberGroup = match.Groups["number"];
+            if (numberGroup.Success)
             {
-                var match = Regex.Match(semanticVersion.BuildMetaData.Branch, configuration.LabelNumberPattern);
-                var numberGroup = match.Groups["number"];
-                if (numberGroup.Success)
-                {
-                    // TODO: Why do we manipulating the semantic version here in the VariableProvider? The method name is GET not MANIPULATE.
-                    // What is about the separation of concern and single-responsibility principle?
-                    semanticVersion.PreReleaseTag.Name += numberGroup.Value;
-                }
+                // TODO: Why do we manipulating the semantic version here in the VariableProvider? The method name is GET not MANIPULATE.
```

Comment:
I wonder if these TODOs are still relevant taking in consideration we compute the semantic version and return it

Reply:
Okay no problem I have removed the comment.